Migrate component to TS: Tag

dhruvv173 opened the pull request:
## Explanation

- This PR fixes #19125 

<!--
Thanks for the pull request. Take a moment to answer these questions so that reviewers have the information they need to properly understand your changes:

* What is the current state of things and why does it need to change?
* What is the solution your changes offer and how does it work?

Are there any issues, Slack conversations, Zendesk issues, user stories, etc. reviewers should consult to understand this pull request better? For instance:

-->

## Screenshots/Screencaps

<!-- If you're making a change to the UI, make sure to capture a screenshot or a short video showing off your work! -->

### Before

<!-- How did the UI you changed look before your changes? Drag your file(s) below this line: -->
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/124f584d-a90a-4b8d-bc9e-a178907a9098)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/81346e64-b94a-4f54-8d96-664838d0e090)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/1324540c-5ec5-42ee-887e-ac8cf3bf9bb1)


### After

<!-- How does it look now? Drag your file(s) below this line: -->
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/29399314-d904-423f-b468-dfc0aff98b4d)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/6c54aa86-6b69-4162-8a1f-00c1f723e48b)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/3f8794c9-4696-4593-ba02-f3270808e116)


unit test coverage
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/eb0baf8d-8c30-4e7e-bdc0-6ada1488177d)

## Manual Testing Steps

<!--
How should reviewers and QA manually test your changes? For instance:

- Go to this screen
- Do this
- Then do this
-->

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

dhruvv173 committed:
migrating Tag to TS

dhruvv173 committed:
updating snapshot

dhruvv173 committed:
type updates

dhruvv173 commented on ui/components/component-library/tag/tag.types.ts:
```diff
@@ -0,0 +1,27 @@
+export type TagProps<C extends React.ElementType> =
+  PolymorphicComponentPropWithRef<C, TagStyleUtilityProps>;
+
+export type TagComponent = <C extends React.ElementType = 'div'>(
```
is this correct? @georgewrmarshall

dhruvv173 commented on ui/components/component-library/tag/tag.types.ts:
```diff
@@ -0,0 +1,27 @@
+  /**
+   * The label props of the component. Most Text component props can be used
+   */
+  labelProps?: TextProps<'div'>;
```
could you please confirm about this too?

dhruvv173 committed:
Merge branch 'develop' into fix/19125

georgewrmarshall reviewed (commented):
Hey @dhruvv173, I still need to do a full review but it looks like there are some snapshot and lining issues that need resolving?

dhruvv173 committed:
addressing lint and snapshot issues

dhruvv173 commented on ui/components/component-library/tag/tag.types.ts:
```diff
@@ -0,0 +1,27 @@
+  /**
+   * The text content of the Tag component, can either be a string or ReactNode
+   */
+  label?: string | React.ReactNode;
```
think this union is better than any(?)

dhruvv173 commented:
> Hey @dhruvv173, I still need to do a full review but it looks like there are some snapshot and lining issues that need resolving?

addressed the issues in the recent commit, passes all the tests. However, I think we'll also need a review from snaps engineer?

dhruvv173 committed:
updating snap-account-detail/header

garrettbear committed:
Update ui/components/component-library/tag/tag.tsx

garrettbear committed:
Update ui/components/component-library/tag/tag.types.ts

garrettbear committed:
Update ui/components/component-library/tag/tag.tsx

garrettbear committed:
Merge branch 'develop' into fix/19125

garrettbear committed:
Update ui/components/component-library/tag/tag.stories.tsx

garrettbear committed:
Update ui/components/component-library/tag/tag.stories.tsx

garrettbear committed:
Update ui/components/component-library/tag/tag.stories.tsx

garrettbear committed:
Update ui/components/component-library/tag/tag.stories.tsx

dhruvv173 commented:
> Looking really good @dhruvv173! I committed my suggestions to your PR, so be sure to pull locally if you end up needing to do any changes. 

Yes, will do. Thank you so much for making the changes!

georgewrmarshall commented on ui/components/component-library/tag/tag.types.ts:
```diff
@@ -0,0 +1,27 @@
+  /**
+   * The label props of the component. Most Text component props can be used
+   */
+  labelProps?: TextProps<'span'>;
```
I suppose this is fine but would note that this prevents things like `<Tag labelProps={{ as: 'p' }} />`

georgewrmarshall commented on ui/pages/keyring-snaps/snap-account-detail-page/header.tsx:
```diff
@@ -86,7 +87,7 @@ export const SnapDetailHeader = ({
                 }}
                 className=""
-                height={2}
+                height={BlockSize.TwoTwelfths}
```
This seems like a mistake and it would get ridden over anyway? I think it could be removed

<img width="1132" alt="Screenshot 2023-07-21 at 10 45 29 AM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/b80d2fae-f0cc-4b50-8e99-f07485d2796b">

georgewrmarshall replied:
Do we need to remove `height: auto` from the component?

georgewrmarshall replied:
I think we remove `height={BlockSize.TwoTwelfths}` here to get this PR in. We can review the styles in a different issue

georgewrmarshall reviewed (commented):
Left one comment, we may want to remove some of the styling on he Tag component in future so we can use the style utility props more

garrettbear committed:
fix tag provided to TextProps

dhruvv173 committed:
lint fix

garrettbear committed:
remove height from Tag

garrettbear committed:
Merge branch 'develop' into fix/19125

georgewrmarshall reviewed (approved):
LGTM! Thanks for your contribution @dhruvv173 and for supporting @garrettbear

dhruvv173 commented:
thanks alot for assisting @garrettbear :)
